File: ui/admin-portal/src/app/services/education-level.service.ts
/*
 * Copyright (c) 2024 Talent Catalog.
 *
 * This program is free software: you can redistribute it and/or modify it under
 * the terms of the GNU Affero General Public License as published by the Free
 * Software Foundation, either version 3 of the License, or any later version.
 *
 * This program is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
 * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License
 * for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. If not, see https://www.gnu.org/licenses/.
 */

import {Injectable} from '@angular/core';
import {environment} from "../../environments/environment";
import {HttpClient} from "@angular/common/http";
import {EducationLevel} from "../model/education-level";
import {Observable} from "rxjs";
import {SearchResults} from "../model/search-results";
import {User} from "../model/user";
import {Candidate} from "../model/candidate";
import {SystemLanguage} from "../model/language";

@Injectable({
  providedIn: 'root'
})
export class EducationLevelService {

  private apiUrl: string = environment.apiUrl + '/education-level';

  constructor(private http: HttpClient) { }

  listEducationLevels(): Observable<EducationLevel[]> {
    return this.http.get<EducationLevel[]>(`${this.apiUrl}`);
  }

  search(request): Observable<SearchResults<EducationLevel>> {
    return this.http.post<SearchResults<EducationLevel>>(`${this.apiUrl}/search`, request);
  }

  get(id: number): Observable<EducationLevel> {
    return this.http.get<EducationLevel>(`${this.apiUrl}/${id}`);
  }

  create(details): Observable<EducationLevel>  {
    return this.http.post<EducationLevel>(`${this.apiUrl}`, details);
  }

  update(id: number, details): Observable<EducationLevel>  {
    return this.http.put<EducationLevel>(`${this.apiUrl}/${id}`, details);
  }

  delete(id: number): Observable<boolean>  {
    return this.http.delete<boolean>(`${this.apiUrl}/${id}`);
  }

  addSystemLanguageTranslations(langCode: string, file: File): Observable<SystemLanguage> {
    const formData: FormData = new FormData();
    formData.append('file', file);
    return this.http.post<SystemLanguage>(`${this.apiUrl}/system/${langCode}`, formData);
  }

}

File: server/src/main/java/org/tctalent/server/service/db/EducationLevelService.java
/*
 * Copyright (c) 2024 Talent Catalog.
 *
 * This program is free software: you can redistribute it and/or modify it under
 * the terms of the GNU Affero General Public License as published by the Free
 * Software Foundation, either version 3 of the License, or any later version.
 *
 * This program is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
 * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License
 * for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. If not, see https://www.gnu.org/licenses/.
 */

package org.tctalent.server.service.db;

import java.util.List;

import org.springframework.data.domain.Page;
import org.tctalent.server.exception.EntityExistsException;
import org.tctalent.server.exception.EntityReferencedException;
import org.tctalent.server.model.db.EducationLevel;
import org.tctalent.server.request.education.level.CreateEducationLevelRequest;
import org.tctalent.server.request.education.level.SearchEducationLevelRequest;
import org.tctalent.server.request.education.level.UpdateEducationLevelRequest;

public interface EducationLevelService {

    List<EducationLevel> listEducationLevels();

    Page<EducationLevel> searchEducationLevels(SearchEducationLevelRequest request);

    EducationLevel getEducationLevel(long id);

    EducationLevel createEducationLevel(CreateEducationLevelRequest request) throws EntityExistsException;

    EducationLevel updateEducationLevel(long id, UpdateEducationLevelRequest request) throws EntityExistsException ;

    boolean deleteEducationLevel(long id) throws EntityReferencedException;

}
